onnxmltools/convert/coreml/operator_converters/neural_network/Activation.py

```diff
@@ -55,7 +56,15 @@ def convert_activation(scope, operator, container):
     else:
         raise TypeError('Unsupported activation layer {0}'.format(activation_type))
 
-    container.add_node(op_type, inputs, outputs, **attrs)
+    # For some ONNX versions, some activation operators include a legacy optimization attribute.
+    op_version = 1
+    if op_type in ['LeakyRelu', 'ReLU', 'PReLU', 'ELU', 'Tanh', 'Sigmoid', 'HardSigmoid']:
+        if container.targeted_onnx_version < StrictVersion('1.0'):
+            attrs['consumed_inputs'] = [0]
+        else:
+            op_version = 6
+
+    container.add_node(op_type, inputs, outputs, op_version=op_version, **attrs)
 
 
```

Comment:
This looks like you are breaking the logic of moving these to _apply_operation. However I like this much better. I think you should define these as a method here in each op when possible.

Reply:
If we put version-depedent code into each converter, it means we will have two copies of the same logic for Core ML and Keras. We need a centralized way to apply ONNX operations shared by all supported packages.